intellij-plugin-structure/structure-edu/src/main/kotlin/com/jetbrains/plugin/structure/edu/EduPlugin.kt

```diff
@@ -27,4 +30,53 @@ data class EduPlugin(
   override val changeNotes: String? = null
 
   val parsedEduVersion = EduPluginVersion.parse(eduPluginVersion)
-}
\ No newline at end of file
+}
+
+data class EduStat(
+  val sections: List<String>,
+  val lessons: List<String>,
+  val tasks: Map<String, Int>
+) {
+
+  fun countInteractiveChallenges(): Int {
+    val ideTasks = tasks[TaskType.IDE.id] ?: 0
+    val outputTasks = tasks[TaskType.OUTPUT.id] ?: 0
+    val eduTasks = tasks[TaskType.EDU.id] ?: 0
+    return ideTasks + outputTasks + eduTasks
+  }
+
+  fun countQuizzes(): Int {
+    return tasks[TaskType.CHOICE.id] ?: 0
+  }
+
+  fun countTheoryTasks(): Int {
+    return tasks[TaskType.THEORY.id] ?: 0
+  }
+
+  companion object {
+    fun fromDescriptor(descriptor: EduPluginDescriptor): EduStat {
+      val allItems = mutableListOf<EduItem>()
+      allItems += descriptor.items
+      descriptor.items.flatMapTo(allItems, { it.items })
```

Comment:
why do you need to do this? 

why not just `descriptor.items.filter {} `

Reply:
There are 2 types of top-level items in the course -- sections and lessons. And sections, in turn, could contain lessons. Therefore I need to collect all the items before filtering.

Reply:
maybe `val all = descriptor.items.flatMap { it.items } + descriptor.items` ?